scripts/extensions/sams/src/utils/redux.ts

```diff
@@ -0,0 +1,55 @@
+// External Modules
```

Comment:
I appreciate your effort in converting this, but I don't think it's simpler than the previous version. It does remove some boilerplate but adds complexity, type safety isn't as strong and interfaces look a bit weird, at least to me since I don't understand them well.

I wasn't suggesting to merge actions and reducers together, just get rid of one-liner action creator functions and simply inline the actions like:


```
interface IIncrementAction {
    type: 'INCREMENT';
    amount: number;
}

interface IDecrementAction {
    type: 'DECREMENT';
    amount: number;
}

interface IResetAction {
    type: 'RESET';
}

type IAction = IIncrementAction | IDecrementAction | IResetAction;

function counter(state, action: IAction) {
    if (typeof state === 'undefined') {
        return 0;
    }

    switch (action.type) {
        case 'INCREMENT':
            return state + action.amount;
        case 'DECREMENT':
            return state - action.amount;
        case 'RESET':
            return 0;
        default:
            return assertNever(action.type);
    }
}

var store = Redux.createStore(counter);

store.dispatch({type: 'DECREMENT', amount: 1});

```

Could you revert this action-reducer merge and simply use inline actions in simple cases?

Reply:
I have reverted back to my original Redux implementation, as that implementation is based on https://redux.js.org/recipes/usage-with-typescript (a recipe for using Redux with Typescript by the Redux team). 

If any developer wishes to learn how state management is used in this extension, we can point that developer to the Redux website and the above recipe provided.